float32 data1
float64 data2